include/nuttx/spi/slave.h

```diff
@@ -157,6 +161,23 @@
 
 #define SPI_SCTRLR_QFLUSH(c)  ((c)->ops->qflush(c))
 
+/****************************************************************************
+ * Name: SPI_SCTRLR_RXQPOLL
+ *
+ * Description:
+ *   Tell the controller to output all the receive queue data.
+ *
+ * Input Parameters:
+ *   sctrlr - SPI slave controller interface instance
+ *
+ * Returned Value:
+ *   Number of bytes left in the rx queue. If the device accepted all the
+ *   data, the return value will be 0
+ *
+ ****************************************************************************/
+
+#define SPI_SCTRLR_RXQPOLL(c)  ((c)->ops->rxqpoll(c))
```

Comment:
This will cause 1..n SPI_SDEV_RECEIVE calls back to the slave device (offering blocks of data to the device). From each call, the slave device driver will return the number of data bytes it accepted/read out. If the slave device returns that it received less bytes than what was offered to it, the poll will return. So it is up to the slave device:

If the slave device wants the poll to return and leave data into the controller driver's queue it shall return any number less than what was offered. The controller driver will discard as many bytes of data as what slave device returns.
if the slave device wants the poll to return and let the controller driver throw away the buffered data it shall return the same number of bytes that was offered to each receive call.

Reply:
How about we put the above description to header file and reword line 266 to 434?